src/vec_buffers.h

```diff
@@ -651,6 +657,69 @@ void Buffers<T>::swap(unsigned i, unsigned j)
     }
 }
 
+/** Perform a preparatiion in radix-2 FFT algorithm
+ *
+ * @param input - input buffers
+ * @param scrambler - a vector stores bit-reversed values
+ * @param data_len - a conven length
+ * @param group_len - number of elements in which a same operations will be
+ * performed
+ */
+template <typename T>
+void Buffers<T>::radix2_fft_prepare(
+    const Buffers<T>& input,
+    const std::vector<T>& scrambler,
+    unsigned data_len,
+    unsigned group_len)
+{
+    const unsigned input_len = input.get_n();
+    const std::vector<T*>& i_mem = input.get_mem();
+    const std::vector<uint8_t*>& i_meta = input.get_meta();
+
+    for (unsigned idx = 0; idx < input_len; ++idx) {
+        // set output  = scramble(input), i.e. bit reversal ordering
+        for (unsigned i = scrambler[idx]; i < scrambler[idx] + group_len; ++i) {
+            std::copy_n(i_mem[idx], size, mem[i]);
+            m_meta ? std::copy_n(i_meta[idx], meta_size, meta[i]) : nullptr;
+        }
+    }
+    for (unsigned idx = input_len; idx < data_len; ++idx) {
+        // set output  = scramble(input), i.e. bit reversal ordering
+        for (unsigned i = scrambler[idx]; i < scrambler[idx] + group_len; ++i) {
+            std::fill_n(mem[i], size, 0);
+            m_meta ? std::fill_n(meta[i], meta_size, 0) : nullptr;
```

Comment:
Ternary are usually used for assignment. That's not the case here.

Could would be clearer written as:

```
if (m_meta) {
    std::fill_n(meta[i], meta_size, 0)
}
```